app/lib/check_records/dfe_sign_in.rb

```diff
@@ -13,5 +13,53 @@ def self.bypass_active?
     def self.review_app?
       HostingEnvironment.review?
     end
+
+    class Client
+      class InvalidTokenError < StandardError; end
+      class UserNotFoundError < StandardError; end
+
+      TIMEOUT_IN_SECONDS = 5
+
+      attr_reader :token
+
+      def initialize(token:)
+        @token = token
+      end
+
+      def get_user_access_to_service(org_id, user_id)
+        endpoint = "/services/#{service_id}/organisations/#{org_id}/users/#{user_id}"
+        response = client.get(endpoint)
+
+        case response.status
+        when 200
+          response.body
+        when 404
+          raise UserNotFoundError
+        when 401
+          raise InvalidTokenError
+        when 500
+          raise StandardError, "DFE Sign-in API returned 500 #{response.body}}"
+        end
+      end
+
+      def client
+        @client ||=
+          Faraday.new(
+            url: ENV.fetch("DFE_SIGN_IN_API_BASE_URL"),
+            request: {
+              timeout: TIMEOUT_IN_SECONDS
+            }
+          ) do |faraday|
+            faraday.request :authorization, "Bearer", token
```

Comment:
This isn't right,  we need JWT Auth here https://github.com/DFE-Digital/login.dfe.public-api#get-user-access-to-service